### PR TITLE
Feat: Exotic Armor Identifier

### PR DESCRIPTION
**Pull request by StyStatic**
- Added a setting that shows whether an item is exotic or dyed using another method

### CODE REVIEW

**Review by LifeIsAParadox — CHANGES_REQUESTED** [review]
Thank you for your pull request. I like the idea, but I have a few suggestions for improvement.

I've added hysky.de/api/color, which reflects the colors ({"ARACHNE_BOOTS":"139,0,0", ...}).

Next time, please create a branch, as it's not advisable to make changes directly to your master branch.

I also suggest merging the newest changes from our master, as there have been made updates in the priceinfotooltip with #297.

I'm also not sure if it's a good idea to display so much information in the tooltip when the expected and given colors are the same, as there's already a lot of information on an item in Hypixel SkyBlock.

In theory, it would be sufficient to have something like 'Exotic color (Spooky)' at the very bottom of the bar when the color doesn't match the expected one.

**Comment by StyStatic** [comment]
Thank you for the feedback, I'll get to work.

Regarding the part about the branch, I'm not sure that's something I can fix for this PR but I'll keep it in mind for future PR's

**Comment by StyStatic** [comment]
Going to keep a running list of progress made here:

- [ ] Use Http Class for API Request
- [ ] Request from HySky rather than Hypixel
- [ ] Merge incoming changes
- [ ] Color should only show in tooltip when Advanced Tooltips are shown
- [ ] Account for intended dyed armor I.E (Holly Dye)
- [ ] Remove Expected Color from tooltip
- [ ] Only show exotic type if Advanced Tooltips are disabled

**Comment by StyStatic** [comment]
I made a few mistakes with git causing commits to register multiple times, but all the requested changes have been made + some changes that made the code more readable and more efficient.

File: src/main/java/me/xmrvizzy/skyblocker/SkyblockerMod.java [hunk 1]
```diff
@@ -73,6 +75,11 @@ public void onInitializeClient() {
         WikiLookup.init();
         ItemRegistry.init();
         NEURepo.init();
+        try {
```
Comment (AzureAaron) [inline src/main/java/me/xmrvizzy/skyblocker/SkyblockerMod.java:78]: The try catch block should be moved into the actual file itself, and also shouldn't throw a runtime exception which'd cause a game crash but rather log an error and ideally disable the feature

**Review by AzureAaron — CHANGES_REQUESTED** [review]

File: src/main/java/me/xmrvizzy/skyblocker/skyblock/item/exotic/DownloadColors.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            ColorApiData = JsonParser.parseString(jsonResponse).getAsJsonObject();
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+
+    }
+}
```
Comment (LifeIsAParadox) [inline src/main/java/me/xmrvizzy/skyblocker/skyblock/item/exotic/DownloadColors.java:25]: This is not necessary, you can add hysky.de/api/color here: https://github.com/SkyblockerMod/Skyblocker/blob/d30994ef9a3208980f307d69db5e5cf0d42d10ef/src/main/java/me/xmrvizzy/skyblocker/skyblock/item/PriceInfoTooltip.java#L440

and also consider using https://github.com/AzureAaron/Skyblocker/blob/d30994ef9a3208980f307d69db5e5cf0d42d10ef/src/main/java/me/xmrvizzy/skyblocker/skyblock/item/PriceInfoTooltip.java#L411.

**Review by LifeIsAParadox — CHANGES_REQUESTED** [review]
I don't mean this negatively, but as advice, I won't accept your pull request for now. Please create a new pull request, as the timeline would otherwise appear messed up.

**Comment by LifeIsAParadox** [comment]
As I mentioned in the comment, I will close the pull request to not mess up the timeline, but you can create a new one with the mentioned changes.

**Pull request closed without merging**